README.md

```diff
@@ -332,6 +332,22 @@ Each plugin can be individually configured using the following module properties
   export const autoConfig = { name: 'y' }
   ```
 
+  You can also use a callback function if you need to access the parent instance:
+  ```js
+  export const autoConfig = (fastify) => {
```

Comment:
I think we should pass in the `overrideConfig` that was previously merged in

Reply:
Oops... Sorry...



Would it be ok?

```js
let pluginOptions
  if (typeof pluginConfig === 'function') {
    pluginOptions = function (fastify) {
      return { ...pluginConfig(fastify), ...overrideConfig};
    }
  } else {
    pluginOptions = { ...pluginConfig, ...overrideConfig}
  }
 ```

Reply:
Tests are not broken by this code, but I didn't test to ensure it works for now.